presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisConnector.java

```diff
@@ -0,0 +1,150 @@
+import io.airlift.log.Logger;
+import io.prestosql.spi.connector.Connector;
+import io.prestosql.spi.connector.ConnectorMetadata;
+import io.prestosql.spi.connector.ConnectorRecordSetProvider;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.session.PropertyMetadata;
+import io.prestosql.spi.transaction.IsolationLevel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.prestosql.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static io.prestosql.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Kinesis connector implementation that includes a record set provider.
+ * <p>
+ * The first 3 methods are mandatory, the remaining Connector methods have defaults.
+ * Here a ConnectorRecordSetProvider is applicable.
+ */
+public class KinesisConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(KinesisConnector.class);
+
+    private final KinesisMetadata metadata;
+    private final KinesisSplitManager splitManager;
+    private final KinesisRecordSetProvider recordSetProvider;
+
+    private final ArrayList<PropertyMetadata<?>> propertyList;
+
+    private ArrayList<ConnectorShutdown> shutdownObjects;
+
+    @Inject
+    public KinesisConnector(
+            KinesisMetadata metadata,
+            KinesisSplitManager splitManager,
+            KinesisRecordSetProvider recordSetProvider)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.recordSetProvider = requireNonNull(recordSetProvider, "recordSetProvider is null");
+
+        this.propertyList = new ArrayList<PropertyMetadata<?>>();
+        buildPropertyList();
+
+        this.shutdownObjects = new ArrayList<ConnectorShutdown>();
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+    {
+        return metadata;
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean b)
+    {
+        checkConnectorSupports(READ_COMMITTED, isolationLevel);
+        return KinesisTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorRecordSetProvider getRecordSetProvider()
+    {
+        return recordSetProvider;
+    }
+
+    /**
+     * Return the session properties.
+     *
+     * @return the system properties for this connector
+     */
+    @Override
+    public List<PropertyMetadata<?>> getSessionProperties()
+    {
+        return this.propertyList;
+    }
+
+    /**
+     * Build the list of session properties we support to supply them to Presto.
+     */
+    protected void buildPropertyList()
```

Comment:
Let's structure session properties the way other connectors do it. See `HiveSessionProperties` for a good example. The first part should be all the property names:
```java
private static final String CHECKPOINT_ENABLED = "checkpoint_enabled";
private static final String ITERATION_NUMBER = "iteration_number";
```
The constructor creates the metadata:
```java
private final List<PropertyMetadata<?>> sessionProperties;

@Inject
public class KinesisSessionProperties(KinesisConnectorConfig config)
{
    sessionProperties = ImmutableList.<PropertyMetadata<?>>builder()
            .add(booleanProperty(
                    CHECKPOINT_ENABLED,
                    "Are checkpoints used in this session?",
                    config.isCheckpointEnabled(),
                    false))
            .add(integerProperty(
                    ITERATION_NUMBER,
                    "Checkpoint iteration number",
                    config. getIterationNumber(),
                    false))
            .build();        
}
```
Then we have accessors for the properties:
```java
public static boolean isCheckpointEnabled(ConnectorSession session)
{
    return session.getProperty(CHECKPOINT_ENABLED, Boolean.class);
}

public static int getIterationNumber(ConnectorSession session)
{
    return session.getProperty(ITERATION_NUMBER, Integer.class);
}
```
`KinesisSessionProperties` would be injected into `KinesisConnector` so that the metadata can be returned from `getSessionProperties()`.

Reply:
restructured as instructed